libparsec/crates/account/src/login.rs

```diff
@@ -0,0 +1,163 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::path::PathBuf;
+
+use libparsec_client_connection::{
+    AccountAuthMethod, AnonymousAccountCmds, AuthenticatedAccountCmds, ConnectionError, ProxyConfig,
+};
+use libparsec_types::prelude::*;
+
+use super::{
+    Account, AUTH_METHOD_ID_DERIVATION_UUID, AUTH_METHOD_MAC_KEY_DERIVATION_UUID,
+    AUTH_METHOD_SECRET_KEY_DERIVATION_UUID,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum AccountLoginWithPasswordError {
+    #[error("Server provided an invalid password algorithm config: {0}")]
+    BadPasswordAlgorithm(CryptoError),
+    #[error("Cannot communicate with the server: {0}")]
+    Offline(#[from] ConnectionError),
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum AccountLoginWithMasterSecretError {
+    #[error("Cannot communicate with the server: {0}")]
+    Offline(#[from] ConnectionError),
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<AccountLoginWithMasterSecretError> for AccountLoginWithPasswordError {
+    fn from(value: AccountLoginWithMasterSecretError) -> Self {
+        match value {
+            AccountLoginWithMasterSecretError::Offline(err) => {
+                AccountLoginWithPasswordError::Offline(err)
+            }
+            AccountLoginWithMasterSecretError::Internal(err) => {
+                AccountLoginWithPasswordError::Internal(err)
+            }
+        }
+    }
+}
+
+pub(super) async fn account_login_with_master_secret(
+    config_dir: PathBuf,
+    proxy: ProxyConfig,
+    addr: ParsecAddr,
+    auth_method_master_secret: KeyDerivation,
+    // We must recycle anonymous_cmds if it exists, otherwise its drop will complain
+    // during tests if some request registered with `test_register_sequence_of_send_hooks`
+    // has yet to occur.
+    anonymous_cmds: Option<AnonymousAccountCmds>,
+    // Since human handle is supposed to be fetched from the server, this parameter is
+    // always set to `None`... Unless for `Account::test_new` where we precisely want
+    // to avoid this initial server request!
+    human_handle: Option<HumanHandle>,
```

Comment:
Maybe something like that would work,if we wanna be nit-picky
```suggestion
    #[cfg(test)]
    human_handle: HumanHandle,
```
and latter 
```rust
let human_handle = if cfg!(test) {
    human_handle
} else {
    // do the request
};
```